### PR TITLE
Airlock Overlays

### PR DESCRIPTION
**Pull request by xxalpha**
This changes the way airlock icons work, from flick()'ing whole sprites to overlaying images. All elements have been separated so it's easier to sprite new airlocks.

 - Airlock overlays done
 - Airlock construction icons done
 - Airlock painter done

All sprite work done by @WJohn.

### CODE REVIEW

**Comment by optimumtact** [comment]
Nice work :+1: 

PS what solved that runtime in the end?

**Comment by xxalpha** [comment]
src. did

I ended up using a global list, that didn't runtime.

**Comment by WJohn** [comment]
Good luck on this, man. Keep asking for help if you're stumped, I'd love to see this added!

**Comment by bustygoku** [comment]
A good way to use this is adding airlock destruction so they aren't harder to deconstruct than R-walls and hulks/mechs can smash through them

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -48,253 +55,23 @@
...
-
-/obj/machinery/door/airlock/glass_virology
-	icon = 'icons/obj/doors/Doorviroglass.dmi'
-	opacity = 0
-	doortype = /obj/structure/door_assembly/door_assembly_viro/glass
-	glass = 1
+	wires = new(src)
+	if(src.closeOtherId != null)
+		spawn (5)
+			for (var/obj/machinery/door/airlock/A in world)
```
Comment (duncathan) [inline code/game/machinery/doors/airlock.dm]: aren't we against looping through world
I'm not sure but I think I've heard people say that before

Comment (Jordie0608) [inline code/game/machinery/doors/airlock.dm]: I'm pretty sure there's a 'machines' list of all machinery in the world.

Comment (xxalpha) [inline code/game/machinery/doors/airlock.dm]: Not my code, but I could change it.

Comment (RemieRichards) [inline code/game/machinery/doors/airlock.dm]: There is no could about it, you touched it, it's your duty to fix the ```in world``` now.
There's a machine list for definite, but there might also be an airlock one, not sure.

Comment (phil235) [inline code/game/machinery/doors/airlock.dm]: Yep, you can use `in airlocks`

**Comment by xxalpha** [comment]
I am done changing things, everything's done.

**Comment by Core0verload** [comment]
How did you solved overlays flickering problem? Overlays don't support flick, and simply changing icon state means all animations will start from a random frame.

**Comment by xxalpha** [comment]
No flick.

**Comment by Core0verload** [comment]
Then how animations starting from a random frame problem is solved?

**Comment by xxalpha** [comment]
Metastation not compiling because some airlocks are no properly mapped.

**Pull request merged**